features/netsocket/Socket.h

```diff
@@ -17,13 +14,27 @@
  * limitations under the License.
  */
 
+/** @file Socket.h Abstract Socket interface */
+/** @addtogroup netsocket
+ * Mbed OS Socket API
+ * @{ */
+
 #ifndef SOCKET_H
 #define SOCKET_H
 
 #include "netsocket/SocketAddress.h"
 #include "Callback.h"
 
-/** Abstract socket class
+/** Socket interface.
+ *
+ * This class defines the Mbed OS Socket API.
+ * Socket is an abstract interface for communicating to remote endpoints.
+ *
+ * This API is intented to use for applications and libraries instead of
+ * using some protocol specific implementation of it. For example TCPSocket
+ * and UDPSocket are implementations of Socket interface.
```

Comment:
Socket API is intentionally protocol-nonspecific, and allows all protocols to provide the same API, regardless of the underlying transport mechanism.